wasm/validation.cpp

```diff
@@ -682,6 +686,33 @@ tl::expected<void, ValidationError> validate(Module const &m) {
         }
     }
 
+    // https://webassembly.github.io/spec/core/valid/modules.html#start-function
+    if (m.start_section.has_value()) {
+        // Ensure function and type sections exist
+        if (!m.function_section.has_value()) {
+            return tl::unexpected{ValidationError::FunctionSectionUndefined};
+        }
+
+        if (!m.type_section.has_value()) {
+            return tl::unexpected{ValidationError::TypeSectionUndefined};
```

Comment:
This branch isn't reached in the test intended to test it. Some earlier check must fail before we get here in the test intended to check this. Maybe we should (in the future, in a separate PR) return more detailed error info so you know what section validation failed at if it fails. :P

I think this is reachable if you have an empty function section when calling validation, so the test should be updated so this branch is hit.